test/end-to-end/offer.js

```diff
@@ -1624,4 +1648,321 @@ describe("Offer endpoint tests", () => {
         });
 
     });
+
+    describe("POST /offers/:offerId/disable", () => {
+        let test_offer, hidden_default_test_offer, hidden_user_test_offer;
+        beforeAll(async () => {
+            test_offer = await Offer.create({
+                ...generateTestOffer(),
+                owner: test_company._id.toString(),
+                ownerName: test_company.name,
+            });
+
+            hidden_default_test_offer = await Offer.create({
+                ...generateTestOffer({ isHidden: true }),
+                owner: test_company._id.toString(),
+                ownerName: test_company.name,
+            });
+
+            hidden_user_test_offer = await Offer.create({
+                ...generateTestOffer(),
+                owner: test_company._id.toString(),
+                ownerName: test_company.name,
+            });
+
+            await (new OfferService()).disable(hidden_user_test_offer._id, OfferConstants.HiddenOfferReasons.company);
+        });
+
+        test("should fail if not logged in", async () => {
+            await test_agent
+                .del("/auth/login");
+
```

Comment:
This will fail if this test runs before some other logs the user. logout operation should be idempotent in order to prevent this. Created #126 to track this.